### DISCUSSION TITLE
import custom from ttf

### DISCUSSION THREAD

**Discussion by Patrone411** [cduck/drawSvg#118] [Q&A] [answered]
hey there, 
is there a way for me to import fonts that are not available on google fonts from a ttf font file format or something similar?
cheers,

patrick

**Comment by cduck** [comment, marked as answer]
Yes, but you'll need to write a bit more code.  Here's a rough sketch of what you need:
- Copy, with modifications, the CSS from an SVG with an embedded font (like https://github.com/cduck/drawsvg/blob/651c36b7a232b74a0246765589d3519780eb45c1/examples/font.svg#L4-L10 starting from `@font`)
- Read your font file in Python as a `bytes` object (TTF, WOFF, or WOFF2.  I suggest the WOFF format for best browser compatibility.)
- Encode the bytes as a data URL like this: https://github.com/cduck/drawsvg/blob/651c36b7a232b74a0246765589d3519780eb45c1/drawsvg/font_embed.py#L13
- Insert this data URL into the CSS from above.
- Use `Drawing.append_css()` to add the font.
- Use the font in the drawing.
